File: server/src/main/java/com/chatroom/server/ServerGUI.java
package com.chatroom.server;

import javax.swing.*;
import java.awt.BorderLayout;

public class ServerGUI extends JFrame implements ServerDisplay {
    private final JTextArea displayWindow = new JTextArea();

    public ServerGUI() {
        super("Server");
        initComponents();
    }

    private void initComponents() {
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setSize(500, 500);
        add(new JScrollPane(displayWindow), BorderLayout.CENTER);
        setVisible(true);
    }

    @Override
    public void showMessage(String message) {
        SwingUtilities.invokeLater(() -> displayWindow.append(message));
    }
}

File: server/src/main/kotlin/com/chatroom/server/ServerGUI.kt
package com.chatroom.server

import javax.swing.*
import java.awt.BorderLayout

class ServerGUI : JFrame("Server"), ServerDisplay {
    private val displayWindow: JTextArea = JTextArea()

    init {
        defaultCloseOperation = EXIT_ON_CLOSE
        setSize(500, 500)
        add(JScrollPane(displayWindow), BorderLayout.CENTER)
        isVisible = true
    }

    override fun showMessage(message: String) {
        SwingUtilities.invokeLater {
            displayWindow.append(message)
        }
    }
}
